### DISCUSSION TITLE
[Help]  Is it possible to use absolute paths when running ZkNotes with the option `hrefs`?

### DISCUSSION THREAD

**Discussion by asterixpound** [zk-org/zk-nvim#152] [Help] [answered]
### Verify

- [X] I searched the existing discussions for help

### How can we help you?

Is there a way to run `:ZkNew` with results filtered to include only those in a given absolute path? The option `hrefs` works for me when I want a relative path. For instance, running

```
:ZkNotes { hrefs = {'fleeting'}}
```

from the root directory of my notebook at `~/Notebook` shows me the notes in the subdirectory `~/Notebook/fleeting`. However, suppose I initially have a file `~/Notebook/permanent/note.md` open in Neovim. Then running the command above looks for notes in `~/Notebook/permanent/fleeting`.

I could use a relative path and run

```
:ZkNotes { hrefs = {'../fleeting'}}
```

but ultimately I am interested in having a keybinding which looks for notes in `~/Notebook/fleeting` regardless of where I start out.

Providing an absolute path as with

```
:ZkNotes { hrefs = {'~/Notebook/fleeting'}}
```

just shows all the notes in my Notebook.

I really appreciate any help on this. I have been finding `zk-nvim` to be an excellent tool which is increasingly becoming an essential part of my work flow.

**Comment by tjex** [comment, marked as answer]
Hey there. `:ZkNotes` calls the `pick_notes()` function, [which always queries from notebook root](https://github.com/zk-org/zk-nvim/blob/fb0962b75a680561f94cae0588b8da92ea8d2fae/lua/zk.lua#L102). It is not dependant on where you are when you execute. 

So, the behaviour you're looking for is already possible. 

Given the following tree:

```tree
.
├── bo4m.md
├── foo
│   ├── bar
│   │   └── ao96.md
│   └── keiv.md
```

no matter where you are in your notebook:
`ZkNotes { hrefs = {"foo"}}` : returns `ao96.md` and `keiv.md`
`ZkNotes { hrefs = {"foo/bar"}}` : returns `ao96.md` only

**Comment by asterixpound** [comment]
Thanks @tjex. Things seem to be working as expected now. I am not sure what what was going on before but it must have been an error on my end.
